src/PrestaShopBundle/EventListener/Context/API/ApiLegacyContextListener.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShopBundle\EventListener\Context\API;
+
+use Symfony\Component\HttpKernel\Event\RequestEvent;
+
+class ApiLegacyContextListener
+{
+    public function __construct(
+        private readonly iterable $legacyBuilders
+    ) {
+    }
+
+    public function onKernelRequest(RequestEvent $event): void
+    {
+        foreach ($this->legacyBuilders as $legacyBuilder) {
```

Comment:
```suggestion
    {
        if (!$event->isMainRequest() || !$this->isResourceApiRequest($event->getRequest())) {
            return;
        }
        
        foreach ($this->legacyBuilders as $legacyBuilder) {
```

Reply:
You must check that the request is an API request, this listener must not be executed in the BO/Admin context which is why we need to perform this check In the PR I'm working on both environments will be correctly separated and listeners are configured to be executed in the appropriate application so the check won't be needed any more

But for now the listeners are common between the Admin and the API although they MUST not all be executed that's why we need this check for now

Note: the `isResourceApiRequest` is defined in the `ExternalApiTrait` that you need to include in this class